.github/workflows/bundle_with_dakota_macos.yml

```diff
@@ -172,14 +180,23 @@ jobs:
 
         echo "libgccpath=$libgccpath"
         echo "libdakotapath=$libdakotapath"
-
-        DYLD_LIBRARY_PATH="$DYLD_LIBRARY_PATH:$libgccpath:libdakotapath"
+        
+        # This may seem like nonsense but is an intentional
+        # workaround MacOS (flaky) security features that result
+        # in delocate suddenly finding the search paths to be empty
+        # ref issue: https://github.com/pypa/cibuildwheel/issues/816
+        EXTENDED_DYLD_LIBRARY_PATH="${libgccpath}:${HOME}/${INSTALL_DIR}/lib:$INSTALL_DIR/bin"
 
         install_name_tool -add_rpath "$libgccpath" "$carolina_so_path"
         install_name_tool -add_rpath "${HOME}/${INSTALL_DIR}/lib" "$carolina_so_path"
 
         mkdir /tmp/carolina_dist
         mkdir /tmp/carolina_dist_unfixed
+        echo "which pip: $(which pip)"
+        echo "which python: $(which python)"
+        echo "$(python -c "import platform; mac_version=platform.mac_ver(); print('macOS version from py: '+mac_version[0])")"
+        pip debug -v|grep -v macosx_10_
```

Comment:
Why would you avoid grepping for `macosx_10_` ?
I understand this might be part of the troubleshooting but am curious nonetheless.

Reply:
This is now hard for me to justify, other than just trying some stuff and it was part of a bigger copypaste. We use macos versions 11 and up ideally